apps/foundation-demo/src/app/core/image-gallery/components/image-gallery.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, Input, ViewEncapsulation } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Store, select } from '@ngrx/store';
+
+import * as fromFoundationImageGallery from '../reducers/index';
+import { SetSelectedImageState } from '../actions/image-gallery.actions';
+
+@Component({
+  selector: 'image-gallery-container',
+  templateUrl: './image-gallery.component.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class ImageGalleryComponent {
+
+  @Input() images;
```

Comment:
Basically, it's because the images aren't actually part of the Product model, and they probably should be, I think. When I originally made up the Product model and the images, I just hard coded the image urls in the product-component. So they aren't in state, because they should really be coming back with the getProducts/getProduct call, and that's not how I did it the first time around. Could I add an issue to move the retrieval of Images into a service call?

Reply:
Yes.